overlord/snapstate/backend/setup.go

```diff
@@ -267,11 +263,16 @@ func (b Backend) RemoveSnapDir(s snap.PlaceInfo, hasOtherInstances bool) error {
 
 func (b Backend) RemoveComponentDir(cpi snap.ContainerPlaceInfo) error {
 	compMountDir := cpi.MountDir()
-	// Remove /snap/<snap_instance>/components/<snap_rev>/<comp_name>
+	// Remove /snap/<snap_instance>/components/mnt/<snap_rev>/<comp_name>/<comp_rev>
 	os.Remove(compMountDir)
-	// and /snap/<snap_instance>/components/<snap_rev> (might fail
-	// if there are other components installed for this revision)
+	// and /snap/<snap_instance>/components/mnt/<snap_rev>/<comp_name> (might fail
+	// if there are other revisions for this component)
 	os.Remove(filepath.Dir(compMountDir))
+
+	// TODO should we check here if there are other components installed
+	// for this snap revision or for other revisions and if not delete
+	// <snap_rev>/ and maybe also components/mnt/<snap_rev>/?
```

Comment:
mnt/<snap_rev> shouldn't exist

Reply:
Right - I've in the end move forward with this and make sure that subdirectories are removed if empty.